File: parking-system/BE/PBMSystem/PBMSystem/Repositories/Entities/ParkingSession.cs
using System;
using Repositories.Helpers;

namespace Repositories.Entities;

public class ParkingSession : BaseEntity
{
    /// <summary>The authenticated user who owns this session. Null for walk-in/anonymous sessions.</summary>
    public Guid? UserId { get; set; }
    public User? User { get; set; }

    public string LicensePlate { get; set; } = string.Empty;
    public string QrCode { get; set; } = string.Empty;
    public string? EntryPhoto { get; set; }
    public string? ExitPhoto { get; set; }
    public DateTime EntryTime { get; set; } = VietnamTime.Now;
    public DateTime? ExitTime { get; set; }
    public string Status { get; set; } = "Active"; // "Active" or "Completed"
    public string? ExitLicensePlate { get; set; }
    public bool? IsPlateMatched { get; set; }
    public bool? IsCheckedIn { get; set; } = false;

    // Reservation fields
    public Guid? ParkingLotId { get; set; }
    public ParkingLot? ParkingLot { get; set; }
    public string? ParkingLotName { get; set; }
    public string? VehicleType { get; set; }
    public string? ReservationDate { get; set; }
    public string? ReservationStartTime { get; set; }
    public string? ReservationEndTime { get; set; }
    public string? ReservationEndDate { get; set; }
    public string? ParkingSlot { get; set; }

    public bool? IsReminderSent { get; set; } = false;

    public System.Collections.Generic.ICollection<ParkingSessionSurcharge> Surcharges { get; set; } = new System.Collections.Generic.List<ParkingSessionSurcharge>();
}

File: parking-staff/src/types/ParkingSession.ts
export interface ParkingSession {
  plate: string;
  status: string;
  time: string;
  createdTimeStr: string;
  createdDateStr: string;
  entryTimeStr: string;
  entryDateStr: string;
  exitTimeStr: string;
  exitDateStr: string;
  isCheckedIn: boolean;
  type: 'ENTRY' | 'EXIT' | 'PENDING' | 'CANCELLED' | 'ALERT';
  owner: string;
  ticketType: string;
  customerName: string | null;
  customerPhone: string | null;
  customerEmail: string | null;
  photo: string;
  entryPhoto?: string;
  exitPhoto?: string;
  qrCode: string;
  totalFee: number;
  parkingLotName?: string;
  parkingSlot?: string;
}

export interface ParkingSessionRaw {
  id?: string;
  licensePlate: string;
  entryTime: string;
  exitTime?: string;
  createdAt?: string;
  status: 'Active' | 'Completed' | 'Cancelled' | string;
  isCheckedIn: boolean;
  userId?: string;
  user?: {
    firstName?: string;
    lastName?: string;
    email?: string;
    phoneNumber?: string;
  };
  entryPhoto?: string;
  exitPhoto?: string;
  qrCode: string;
  totalFee: number;
  parkingLotName?: string;
  parkingLotId?: string;
  ParkingLotId?: string;
  parkingSlot?: string;
  vehicleType?: string;
  VehicleType?: string;
  reservationDate?: string;
  reservationStartTime?: string;
}

export interface GateScanPayload {
  qrCode: string;
  entryPhoto: string;
}

export interface CheckinPayload {
  licensePlate: string;
  entryPhoto: string;
  parkingLotName: string;
  parkingLotId?: string;
  vehicleType: string;
}

export interface CheckoutPayload {
  qrCode: string;
  exitLicensePlate: string;
  exitPhoto: string;
  extraFees: { name: string; amount: number }[];
}

export interface VerifyQrResponse {
  session: ParkingSessionRaw;
  Session?: any;
  user?: {
    firstName?: string;
    lastName?: string;
    email?: string;
    phoneNumber?: string;
  };
  User?: any;
  fee?: number;
  Fee?: number;
  prepaidAmount?: number;
  PrepaidAmount?: number;
}

export interface BlacklistEntry {
  id?: string;
  plateNumber: string;
  reason: string;
  createdAt?: string;
}
